addons/event_sale/wizard/event_edit_registration.py

```diff
@@ -50,6 +52,19 @@ def default_get(self, fields):
         res = self._convert_to_write(res)
         return res
 
+    @api.depends('event_registration_ids')
+    def _compute_show_event_column(self):
+        for event_edit_registration in self:
+            event_count = self.env['event.registration'].read_group(
+                [
+                    ('sale_order_id', '=', event_edit_registration.sale_order_id.id),
+                    ('event_ticket_id', 'in', event_edit_registration.sale_order_id.mapped('order_line.event_ticket_id').ids),
+                ],
+                ['event_id'], ['event_id'],
+            )
+            #only show event column if there are multiple events
+            event_edit_registration.show_event_column = True if len(event_count) > 1 else False
```

Comment:
Mmmh, is it really how it works?
Aren't you supposed to do something like `read_group_res[0]['event_id_count']` ?

Reply:
Actually what we are trying to achieve here is to get the distinct events count in all the event registration so that if there are more then 1 events, we show the events column in the tree_view.
After having a look at this one, I think that we can simplify this and rather then using the same domain as above to get registrations in the default_get we can calculate the event_count this way;
 `event_count = len(event_edit_registration.event_registration_ids.mapped('event_id').ids)
  #only show event column if there are multiple events
  event_edit_registration.show_event_column = (event_count > 1)`

What do you think about it?

Reply:
That sounds correct and more simple, yes!
`event_edit_registration.show_event_column = len(event_edit_registration.event_registration_ids.event_id) > 1`

I think "mapped" will eliminate duplicate records within the recordset but make sure to double-check.

Reply:
Indeed, changed to `event_edit_registration.show_event_column = len(event_edit_registration.event_registration_ids.event_id) > 1`